lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -380,6 +399,28 @@ class ReportUIFeatures {
     this._document.body.removeChild(a);
     setTimeout(_ => URL.revokeObjectURL(href), 500);
   }
+
+  _updateStickyHeaderOnScroll() {
+    // Show sticky header when the score scale begins to go underneath the topbar.
+    const topbarBottom = this.topbarEl.getBoundingClientRect().bottom;
+    const scoreScaleTop = this.scoreScaleEl.getBoundingClientRect().top;
+    const showStickyHeader = topbarBottom >= scoreScaleTop;
+    this.stickyHeaderEl.classList.toggle('lh-sticky-header--visible', showStickyHeader);
```

Comment:
we'll want to change this at the bottom.

otherwise we force a layout within this function.

![image](https://user-images.githubusercontent.com/39191/56706570-5363ff00-66ca-11e9-8cfb-948bd27bc7f9.png)

Reply:
Interesting... The order is the same, so is the issue that code in between these two changes allows for the browser to jump in and do an extra layout? Idk how layouts work

Reply:
https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing

or http://wilsonpage.co.uk/preventing-layout-thrashing/

Reply:
got it, makes sense.

AFAICT from the performance tab w/ this code, there are no extra layouts. One change I saw was "Recalculate Styles" was within the update function, but w/ your changes it ran just after on the main thread. Would like to know if I'm reading the performance timeline wrong.

Even if there technically isn't a layout cost, moving the DOM mutation to the bottom SGTM b/c it makes it obvious there is no layout thrashing.